src/ClientResolver.php

```diff
@@ -118,6 +118,12 @@ class ClientResolver
             'doc'   => 'Allows you to specify which profile to use when credentials are created from the AWS credentials file in your HOME directory. This setting overrides the AWS_PROFILE environment variable. Note: Specifying "profile" will cause the "credentials" key to be ignored.',
             'fn'    => [__CLASS__, '_apply_profile'],
         ],
+        'config_profile' => [
```

Comment:
Since there is a `profile` parameter already (which checks for credential file only), I thought it could be handy when customer knows clearly what they are doing, they don't need to create a provider. It doesn't overlap with `AWS_PROFILE`, it just make it quicker in deciding which type of file is looking at.

For other SDKs, I know Ruby just has `profile` parameter only, and it use credential chain to check credential file before checking config file.